tests/robot-tests/tests/libs/fail_fast.py

```diff
@@ -46,32 +64,37 @@ def record_failing_test_suite():
 
 def fail_test_fast_if_required():
     if current_test_suite_failing_fast():
-        _raise_assertion_error(f"Test suite {_get_current_test_suite()} is already failing.  Failing this test fast.")
+        raise SkipExecution(f"Test suite {_get_current_test_suite()} is already failing.  Skipping this test.")
 
 
 def get_failing_test_suites() -> []:
-    if os.path.isfile(failing_suites_filename):
-        # We wouldn't expect the same test suite to be recorded in this file more than once, as we only trigger the
-        # "record failing test suite" upon the first failing test in an individual suite.
-        #
-        # Strangely though, this does get called multiple times if using Pabot and re-running failed suites. It seems as
-        # though the failure keywords are being merged with the initial run's failure keyword definitions and therefore
-        # causing the failing test suite to be recorded multiple times when its first failing test is hit.
-        #
-        # We therefore explicitly remove any duplicates from the list here.
-
-        with file_lock:
+    with file_lock:
+        if os.path.isfile(failing_suites_filename):
             try:
                 with open(failing_suites_filename, "r") as file:
-                    failing_suites = file.readlines()
-                    stripped_suite_names = [failing_suite.strip() for failing_suite in failing_suites]
-                    filtered_suite_names = filter(None, stripped_suite_names)
-                    return list(dict.fromkeys(filtered_suite_names))
+                    return [suite.strip() for suite in file.readlines()]
             except IOError as e:
                 logger.error(f"Failed to read failing test suites from file: {e}")
                 return []
-    return []
+        return []
+
+
+def _capture_html():
+    html = sl().get_source()
+    current_time_millis = round(datetime.datetime.timestamp(datetime.datetime.now()) * 1000)
+    output_dir = BuiltIn().get_variable_value('${OUTPUT DIR}')
+    html_file = open(f"{output_dir}{os.sep}captured-html-{current_time_millis}.html", "w", encoding="utf-8")
+    html_file.write(html)
+    html_file.close()
+    logger.info(f"Captured HTML of {sl().get_location()}      HTML saved to file://{os.path.realpath(html_file.name)}")
```

Comment:
**nitpick** Just wondering what the spacing between `{sl().get_location()` to `HTML` is for here? I'm assuming it's for CI but not too sure :)

Reply:
Yeah, it's basically that!  Separate a large HTTP URL from another large file URL in the logs to make it easier to read.